application/AuthTask.cpp

```diff
@@ -0,0 +1,76 @@
+// Licensed under the Apache-2.0 license. See README.md for details.
+
+#include "AuthTask.h"
+
+#include "auth/AccountModel.h"
+#include "auth/BaseAccount.h"
+#include "BaseInstance.h"
+#include "dialogs/AccountLoginDialog.h"
+#include "dialogs/AccountsDialog.h"
+#include "Exception.h"
+#include "MultiMC.h"
+
+AuthTask::AuthTask(const QString &type, InstancePtr instance, SessionPtr session, QObject *parent)
+	: StandardTask(parent), m_type(type), m_instance(instance), m_session(session)
+{
+}
+
+void AuthTask::executeTask()
+{
+	BaseAccount *account = MMC->accountsModel()->get(m_type, m_instance);
+	if (account)
+	{
+		try
+		{
+			setStatus(tr("Checking if default account is valid..."));
+			runTask(account->createCheckTask(m_session));
+			// easiest: have default and it's valid
+			m_account = account;
+			emitSucceeded();
+			return;
+		}
+		catch (...)
+		{
+			setStatus(tr("Querying user for authentication details..."));
+			emit progress(101, 100);
+			AccountLoginDialog dlg(account);
```

Comment:
This task was originally in logic, and used LogicalGui to show stuff. You told me to not use LogicalGui, which required me to put GUI in logic and the entire thing into application/.